api/batches/hooks.js

```diff
@@ -97,9 +97,25 @@ Batches.after.update(
       return;
     }
 
-    [Games, GameLobbies].forEach(coll => {
-      coll.update({ batchId }, { $set: { status } }, { multi: true });
-    });
+    const unfinishedGames = Games.find({
+      batchId,
+      finishedAt: { $exists: false }
+    }).fetch();
+
+    if (unfinishedGames.length > 0) {
+      const unfinishedGameIds = _.pluck(unfinishedGames, "_id");
+
+      Games.update(
+        { batchId, _id: { $in: unfinishedGameIds } },
+        { $set: { status } },
+        { multi: true }
+      );
+      GameLobbies.update(
+        { batchId, gameId: { $in: unfinishedGameIds } },
+        { $set: { status } },
+        { multi: true }
+      );
+    }
```

Comment:
Instead of searching the unfinished games, shouldn't it check for `status: { $ne: "finished" }`, or maybe rather `status: { $nin: [ "finished", "cancelled", "failed", "custom" ] }`.